tests/unit/test_shard_aware.py

```diff
@@ -88,12 +89,12 @@ def mock_connection_factory(self, *args, **kwargs):
                 connection.is_defunct = False
                 connection.is_closed = False
                 connection.orphaned_threshold_reached = False
-                connection.endpoint = args[0]
-                connection.shard_id = kwargs.get('shard_id', self.connection_counter)
+                connection.endpoint = args[0] 
+                connection.features = ProtocolFeatures(shard_id=kwargs.get('shard_id', self.connection_counter), 
+                                                       sharding_info=ShardingInfo(shard_id=1, shards_count=4,
+                                                                    partitioner="", sharding_algorithm="", sharding_ignore_msb=0,
+                                                                    shard_aware_port=19042, shard_aware_port_ssl=19045))
```

Comment:
Nit: it would be more readable imho if this `ShardingInfo` was created separately and assigned to temporary var. Right now there is too much clutter in this constructor.